### ISSUE TITLE
Crash when trying to get GDrive contents in API 16 emulator

### ISSUE THREAD

**Issue by lognaturel** [getodk/collect#488]
#### Software and hardware versions 
Collect v1.4.16-18-gd398fc05, Android 4.1.2 emulator, Google Play Services 9256030

#### Problem description
Collect crashes when trying to list contents of GDrive.

#### Steps to reproduce the problem
1. Set Platform to Google Drive, Google Sheets in General Settings
1. Tap Get Blank Form
1. Tap My Drive

#### Expected behavior
This seems to be due to the Play Services version. The user should be told that their Play Services is out of date and must be updated to use this feature. They should be given the option to cancel fetching new forms so the app doesn't crash. Perhaps they can be linked to update Play Services.

#### Other information 
I can run the same code in an API 22 emulator with no problems.

This is related to #132.

````
W/GooglePlayServicesUtil: Google Play services out of date.  Requires 10084000 but found 9256030
E/AndroidRuntime: FATAL EXCEPTION: AsyncTask #1
                  java.lang.RuntimeException: An error occured while executing doInBackground()
                      at android.os.AsyncTask$3.done(AsyncTask.java:299)
                      at java.util.concurrent.FutureTask$Sync.innerSetException(FutureTask.java:273)
                      at java.util.concurrent.FutureTask.setException(FutureTask.java:124)
                      at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:307)
                      at java.util.concurrent.FutureTask.run(FutureTask.java:137)
                      at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:230)
                      at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1076)
                      at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:569)
                      at java.lang.Thread.run(Thread.java:856)
                   Caused by: android.content.ActivityNotFoundException: No Activity found to handle Intent { act=android.intent.action.VIEW dat=market://details?id=com.google.android.gms&pcampaignid=gcore_10084000-e-org.odk.collect.android-2029 flg=0x80000 pkg=com.android.vending }
                      at android.app.Instrumentation.checkStartActivityResult(Instrumentation.java:1545)
                      at android.app.Instrumentation.execStartActivity(Instrumentation.java:1416)
                      at android.app.Activity.startActivityForResult(Activity.java:3351)
                      at android.app.Activity.startActivityForResult(Activity.java:3312)
                      at org.odk.collect.android.activities.GoogleDriveActivity$RetrieveDriveFileContentsAsyncTask.doInBackground(GoogleDriveActivity.java:567)
                      at org.odk.collect.android.activities.GoogleDriveActivity$RetrieveDriveFileContentsAsyncTask.doInBackground(GoogleDriveActivity.java:529)
                      at android.os.AsyncTask$2.call(AsyncTask.java:287)
                      at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:305)
                      at java.util.concurrent.FutureTask.run(FutureTask.java:137) 
                      at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:230) 
                      at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1076) 
                      at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:569) 
                      at java.lang.Thread.run(Thread.java:856) 
````

**Comment by Dvik** [comment]
i think this error is independent of play services version and is related to invalid or no API key for accessing the Drive API.
https://github.com/opendatakit/collect#using-apis-for-local-development

**Comment by lognaturel** [comment]
@Dvik I am running the exact same code between the two emulators. Did you try it with Google Play Services < 10084000 and get it to work?

Also look at this line: `Caused by: android.content.ActivityNotFoundException: No Activity found to handle Intent { act=android.intent.action.VIEW dat=market://details?id=com.google.android.gms&pcampaignid=gcore_10084000-e-org.odk.collect.android-2029 flg=0x80000 pkg=com.android.vending }`

**Comment by Dvik** [comment]
Not yet. I am downloading API 16 with no google api. Will let know once done.
Funnily enough, i was getting the same error yesterday when i had not set up the APIs for local development.

**Comment by lognaturel** [comment]
@Dvik Great, it will be very interesting to see your experience.

I think it makes sense that the error would be the same. In either case, the problem is that the Intent can't be handled. In one case it's because of the missing keys and in the other because of the version mismatch. But I'm not sure, your attempted reproduction will tell us!

**Comment by Dvik** [comment]
Sorry for not letting know but i was able to reproduce the crash on API 16 as well as API 22.
The fix will be a simple one and will be similar to how such situations are handled with location.
I'll submit a PR after the location PR has been merged after all the changes.

**Comment by lognaturel** [comment]
This will be addressed in #490.
